tester.go

```diff
@@ -209,16 +212,32 @@ func validateResponseHeaders(test *Test, response *http.Response) []error {
 		}
 	}
 
+	// NotPresentSubfield assertions come in the form of key/value pairs where the key is the header name and the value is
+	// the pattern that we want to confirm does not exist within that header. Here we test that in two steps:
+	//		1. Check if the header, itself exists. If it does not the test automatically passes.
+	//		2. If the header does exist we pass the full list of NotPresentSubfield key/value pairs to the validate function
+	//			which errors if the pattern is matched in the header.
+	npSubfieldAssertions := expectedResponse.Headers.NotPresentSubfields
+	for header := range npSubfieldAssertions {
+		respHeaderValue := response.Header.Get(header)
+		if len(respHeaderValue) > 0 {
+			errors = append(errors, validateResponseHeaderPatterns(response, expectedResponse.Headers.NotPresentSubfields, false, expectFailure)...)
+		}
+	}
```

Comment:
It took me a while to understand the difference between `NotPresentSubfield` and the existing `NotMatching` assertions.

Could I suggest renaming `NotPresentSubfield` to `IfPresentNotMatching` (if header exists, not matching)?

This would help differentiate it with `NotMatching` (expecting header to exist and not matching) more clearly

Reply:
Also
```suggestion
	// IfPresentNotMatching assertions come in the form of key/value pairs where the key is the header name and the value is
	// the pattern that we want to confirm does not exist within that header. Here we test that in two steps:
	//		1. If the header doesn't exists, the test automatically passes.
	//		2. If the header does exist, validate against the not matching assertions.
	ipnmHeaders := expectedResponse.Headers.IfPresentNotMatching
	for header := range ipnmHeaders {
		if len(response.Header.Get(header)) > 0 {
			errors = append(errors, validateResponseHeaderPatterns(response, ipnmHeaders, false, expectFailure)...)
		}
	}
```